### PR TITLE
Add EKS fargate container insights using ADOT configuration file

### PR DESCRIPTION
**Pull request by skyduo**
**Description:** 
AWS EKS is going to enable new container insights metrics for fargate using ADOT, this is the configuration file manifest for setting up the OT components.

**Testing:** 
`make build`, testing the config and could see the metrics in CW

### CODE REVIEW

**Comment by alolita** [comment]
@aws-observability/adot-collector-maintainers do not merge until after Jan release and signoff from @Aneurysm9 and @alolita

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (Aneurysm9) [inline deployment-template/eks/otel-fargate-container-insights.yaml:526]: Needs a specific ADOT Collector version tag.

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (Aneurysm9) [inline deployment-template/eks/otel-fargate-container-insights.yaml:552]: Needs `\n` at EOF.

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (hossain-rayhan) [reply deployment-template/eks/otel-fargate-container-insights.yaml:470]: Let's remove this. It was for testing only.

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (hossain-rayhan) [inline deployment-template/eks/otel-fargate-container-insights.yaml:480]: Let's remove the logging exporter.

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (hossain-rayhan) [reply deployment-template/eks/otel-fargate-container-insights.yaml:526]: Let's use the latest version `v0.15.1`. Please test in house as well. Though we already tested with :latest tag which is the same version I believe.

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (hossain-rayhan) [reply deployment-template/eks/otel-fargate-container-insights.yaml:467]: No, we only export pod level metrics. Container stats will be shipped as raw CloudWatch logs.

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (hossain-rayhan) [reply deployment-template/eks/otel-fargate-container-insights.yaml:64]: I am not sure about this. If it doesn't hurt let's go with this. Otherwise we can remove if insisted.

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (skyduo) [reply deployment-template/eks/otel-fargate-container-insights.yaml:470]: got it

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (skyduo) [reply deployment-template/eks/otel-fargate-container-insights.yaml:480]: sure

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (Aneurysm9) [reply deployment-template/eks/otel-fargate-container-insights.yaml:64]: Remove it if we don't have a good reason to have it.  We should strive for configurations that are secure by default and have very good reasons with supporting documentation for deviating from that position.

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (Aneurysm9) [reply deployment-template/eks/otel-fargate-container-insights.yaml:467]: Through what mechanism?  I don't see a CW logs exporter here.  If these metrics are not being shipped by this pipeline why go through the effort to construct them?

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (skyduo) [reply deployment-template/eks/otel-fargate-container-insights.yaml:552]: got it

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (hossain-rayhan) [reply deployment-template/eks/otel-fargate-container-insights.yaml:64]: @skyduo let's remove it test one more time with the final version (after all the changes).

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (hossain-rayhan) [reply deployment-template/eks/otel-fargate-container-insights.yaml:467]: Maybe I didn't understand your question. 

`awsemf` exporter can send logs and metrics, technically logs in EMF format. In EMF format if we include metric names with dimension, those will show up as CloudWatch Metrics. Other data points will still be sent to CloudWatch as raw CloudWatch logs but not in Metric format. We want only 8 of them to be metrics. This was tested and verified. Hope it clarifies.

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (Aneurysm9) [reply deployment-template/eks/otel-fargate-container-insights.yaml:118]: Since there aren't any anchors, unless the matcher adds implicit start and end anchors (which would be annoying), you can just use `\S`.  That will match if there is any non-whitespace and we don't care about anything else.  It is much simpler and clearer to understand for someone looking at this later.

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (Aneurysm9) [reply deployment-template/eks/otel-fargate-container-insights.yaml:467]: This is non-obvious and should be documented.

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (skyduo) [reply deployment-template/eks/otel-fargate-container-insights.yaml:118]: Hi @Aneurysm9! i have tried a lot of new regex like (.|\s)*, (.|\S)*, [\s\S]*, [.\s]*, and [.\S]* but all of them show some errors, do we force need to change this regex now? since we are going to launch this next week and this will not impact on any other code, could we leave this for a phase II change?

Comment (Aneurysm9) [reply deployment-template/eks/otel-fargate-container-insights.yaml:118]: Did you try simply `\S`?  What errors were produced?

I am concerned about this regex because it appears to be overly complicated and confusing.  It should be simple to resolve and we should resolve it now.

Comment (skyduo) [reply deployment-template/eks/otel-fargate-container-insights.yaml:118]: yeah i tried it will post some strange error like CrashLoopBackOff if i just use `\S`, and the error shows like cannot retrieve the configuration: error loading config file "/conf/adot-collector-config.yaml": unable to read the file /conf/adot-collector-config.yaml: yaml: line 70: found unknown escape character, but the line 70 in my testing config file is line 49 in the pr, obviously no error on that line, and also the original regex works well, thus got stuck here

Comment (Aneurysm9) [reply deployment-template/eks/otel-fargate-container-insights.yaml:118]: Since this is in a string in a YAML file representing another YAML file, I don't think you can use `\S` directly, but must escape the `\`, so something like this:

```yaml
experimental_match_labels: {"container": "\\S", "LaunchType": "fargate"}
```

Comment (hossain-rayhan) [reply deployment-template/eks/otel-fargate-container-insights.yaml:118]: We want to make sure it's not null or non empty or non blank. Does it guaranty it has some kind of value?

Comment (Aneurysm9) [reply deployment-template/eks/otel-fargate-container-insights.yaml:118]: > \S matches any non-whitespace character (equivalent to [^\r\n\t\f\v ])

**Comment by skyduo** [comment]
Hi @Aneurysm9! i just update a new revision based on the comments you left, please take a look, Thanks!

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (skyduo) [reply deployment-template/eks/otel-fargate-container-insights.yaml]: changed to `\\S` and it works

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (skyduo) [reply deployment-template/eks/otel-fargate-container-insights.yaml]: i added a new line at EOF in the latest commit but seems like it still shows the same in the pr

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (Aneurysm9) [reply deployment-template/eks/otel-fargate-container-insights.yaml:466]: This still needs documentation.

**Review by Aneurysm9 — APPROVED** [review]

**Comment by skyduo** [comment]
Hi Anthony! i think for this pod level metrics part we will add some new document in the ADOT doc https://aws-otel.github.io/docs/getting-started/container-insights/eks-prometheus, i will work with Chris to add it.

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (alolita) [reply deployment-template/eks/otel-fargate-container-insights.yaml:466]: @Aneurysm9 +1 can you outline what you want to see in documentation? @skyduo can you sync w @Aneurysm9 to ensure you add the info in the docs :-)

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (alolita) [reply deployment-template/eks/otel-fargate-container-insights.yaml]: is this done?

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (Aneurysm9) [reply deployment-template/eks/otel-fargate-container-insights.yaml:466]: I think a list of what will be sent as metrics and what will be sent as logs.  Or some other way to know as a user what data I should expect to be produced by this configuration and where those data will land.

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (skyduo) [reply deployment-template/eks/otel-fargate-container-insights.yaml]: yeah

File: deployment-template/eks/otel-fargate-container-insights.yaml
Comment (skyduo) [reply deployment-template/eks/otel-fargate-container-insights.yaml:467]: left a short inline comment there and add a link to the new EKS fargate CI doc

**Comment by skyduo** [comment]
Hi @alolita @Aneurysm9 ! i just update the inline comment and link to the ADOT doc, please take a look, thanks!

**Review by alolita — APPROVED** [review]

**Pull request merged**
